presto-main/src/main/java/com/facebook/presto/operator/aggregation/TypedSet.java

```diff
@@ -55,7 +55,7 @@ public TypedSet(Type elementType, int expectedSize)
         this.maxFill = calculateMaxFill(hashSize);
         this.hashMask = hashSize - 1;
 
-        blockPositionByHash.ensureCapacity(hashSize);
+        blockPositionByHash.size(hashSize);
```

Comment:
Initialize `blockPositionByHash` using the constructor that takes initial capacity instead.